### PR TITLE
PYTHON-1307 Remove SONManipulator APIs

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

File: doc/api/pymongo/collection.rst [hunk 1]
```diff
@@ -47,8 +47,8 @@
...
-      .. automethod:: find_raw_batches(filter=None, projection=None, skip=0, limit=0, no_cursor_timeout=False, cursor_type=CursorType.NON_TAILABLE, sort=None, allow_partial_results=False, oplog_replay=False, modifiers=None, batch_size=0, manipulate=True, collation=None, hint=None, max_scan=None, max_time_ms=None, max=None, min=None, return_key=False, show_record_id=False, snapshot=False, comment=None)
+      .. automethod:: find(filter=None, projection=None, skip=0, limit=0, no_cursor_timeout=False, cursor_type=CursorType.NON_TAILABLE, sort=None, allow_partial_results=False, oplog_replay=False, modifiers=None, batch_size=0, collation=None, hint=None, max_scan=None, max_time_ms=None, max=None, min=None, return_key=False, show_record_id=False, snapshot=False, comment=None, session=None, allow_disk_use=None)
```
Comment (behackett) [inline doc/api/pymongo/collection.rst:50]: Did we just forget to document allow_disk_use?

File: pymongo/collection.py [hunk 2]
```diff
@@ -554,15 +554,9 @@ def _legacy_write(self, sock_info, name, cmd, op_id,
...
     def _insert_one(
             self, doc, ordered,
-            check_keys, manipulate, write_concern, op_id, bypass_doc_val,
+            check_keys, write_concern, op_id, bypass_doc_val,
             session):
         """Internal helper for inserting a single document."""
-        if manipulate:
-            doc = self.__database._apply_incoming_manipulators(doc, self)
-            if not isinstance(doc, RawBSONDocument) and '_id' not in doc:
-                doc['_id'] = ObjectId()
```
Comment (behackett) [inline pymongo/collection.py:563]: I guess this was just here for legacy insert()?

File: doc/api/pymongo/collection.rst (hunk 1 above)
Comment (ShaneHarvey) [reply doc/api/pymongo/collection.rst:50]: Yep. I just opened https://jira.mongodb.org/browse/PYTHON-2524 to backport this change to 3.X.

File: pymongo/collection.py (hunk 2 above)
Comment (ShaneHarvey) [reply pymongo/collection.py:563]: Yep, the non-legacy `insert_one` method always adds the `_id` field before calling this method.

**Review by behackett — APPROVED** [review]
Good riddance.

**Review by prashantmital — APPROVED** [review]
🥳

**Pull request merged**